crates/object_detection/src/pose_interpretation.rs

```diff
@@ -0,0 +1,256 @@
+    time_to_reach_kick_position: CyclerState<Duration, "time_to_reach_kick_position">,
+
+    camera_matrices: RequiredInput<Option<CameraMatrices>, "Control", "camera_matrices?">,
+    human_poses: Input<Option<Vec<HumanPose>>, "human_poses?">,
+    ground_to_field: Input<Option<Isometry2<Ground, Field>>, "Control", "ground_to_field?">,
+    expected_referee_position:
+        Input<Option<Point2<Ground>>, "Control", "expected_referee_position?">,
+    fall_state: Input<FallState, "Control", "fall_state">,
+
+    player_number: Parameter<PlayerNumber, "player_number">,
+    keypoint_confidence_threshold:
+        Parameter<f32, "detection.$cycler_instance.keypoint_confidence_threshold">,
+    distance_to_referee_position_threshold:
+        Parameter<f32, "detection.$cycler_instance.distance_to_referee_position_threshold">,
+    foot_z_offset: Parameter<f32, "detection.$cycler_instance.foot_z_offset">,
+    shoulder_angle_threshold: Parameter<f32, "detection.$cycler_instance.shoulder_angle_threshold">,
+
+    detected_pose_types: AdditionalOutput<Vec<(PoseType, Point2<Field>)>, "detected_pose_types">,
+}
+
+#[context]
+#[derive(Default)]
+pub struct MainOutputs {
+    pub detected_referee_pose_type: MainOutput<PoseType>,
+}
+
+impl PoseInterpretation {
+    pub fn new(_context: CreationContext<impl PathsInterface>) -> Result<Self> {
+        Ok(PoseInterpretation {})
+    }
+
+    pub fn cycle(
+        &mut self,
+        mut context: CycleContext<impl NetworkInterface>,
+    ) -> Result<MainOutputs> {
+        let (Some(expected_referee_position), Some(human_poses)) =
+            (context.expected_referee_position, context.human_poses)
+        else {
+            context.detected_pose_types.fill_if_subscribed(Vec::new);
+            return Ok(MainOutputs::default());
+        };
+
+        let referee_pose = Self::get_referee_pose(
+            human_poses,
+            context.camera_matrices.top.clone(),
+            *context.distance_to_referee_position_threshold,
+            *expected_referee_position,
+            *context.foot_z_offset,
+        );
+
+        let pose_type = Self::interpret_pose(
+            &referee_pose,
+            *context.keypoint_confidence_threshold,
+            *context.shoulder_angle_threshold,
+        );
+
+        if let PoseType::OverheadArms = pose_type {
+            if let Some(ground_to_field) = context.ground_to_field {
+                context
+                    .hardware_interface
+                    .write_to_network(OutgoingMessage::Spl(HulkMessage {
+                        player_number: *context.player_number,
+                        fallen: matches!(context.fall_state, FallState::Fallen { .. }),
+                        pose: ground_to_field.as_pose(),
+                        over_arms_pose_detected: true,
+                        ball_position: None,
+                        time_to_reach_kick_position: Some(*context.time_to_reach_kick_position),
+                    }))?;
```

Comment:
I think it is not a good idea to let a detection module send messages. IMHO this should be done in control after filtering etc.

Reply:
Done, is now moved to `control::RefereePoseDetectionFilter`.